docs/internals/optimizer.rst

```diff
@@ -41,11 +41,11 @@ often done because it results in opportunities for more simplifications.
 Differences between Optimized and Non-Optimized Code
 ====================================================
 
-Generally, the most visible difference would be constant expressions getting evaluated.
-When it comes to the ASM output, one can also notice reduction of equivalent/duplicate
-"code blocks" (compare the output of the flags ``--asm`` and ``--asm --optimize``). However,
+Generally, the most visible difference is that constant expressions are evaluated at compile time.
+When it comes to the ASM output, one can also notice a reduction of redundant
```

Comment:
Redundant might be too strong here. I think this line is referring to Block Deduplicator. This optimizer step would look for code blocks that are the same, removes all except one and replace jumps to blocks that were removed to the one block that remains.